request_tracing.go

```diff
@@ -39,7 +46,8 @@ func TracingMiddleware(h http.HandlerFunc) http.HandlerFunc {
 			defer func() {
 				if requestID := requestWithContext.Context().Value(ctxRequestIDKey); requestID != nil {
 					if requestIDStr, ok := requestID.(string); ok {
-						w.Header().Set(headerRequestIDKey, requestIDStr)
+						setRequestID(w, headerRequestIDKey, requestIDStr)
```

Comment:
Yep! For now, I want to leave it to ensure compatibility, but probably we can remove the `x-request-id` header